src/main/java/uz/uzfundit/Exercism/QueenAttackCalculator.java

```diff
@@ -0,0 +1,74 @@
+package uz.uzfundIT.Exercism;
+public class QueenAttackCalculator {
+    private final Queen whiteQueen;
+    private final Queen blackQueen;
+
+    QueenAttackCalculator(Queen whiteQueen, Queen blackQueen) {
+        if(whiteQueen == null || blackQueen == null) {
+            throw new IllegalArgumentException("You must supply valid positions for both Queens.");
+        } else if(whiteQueen.equals(blackQueen)) {
+            throw new IllegalArgumentException("Queens cannot occupy the same position.");
+        } else {
+            this.whiteQueen = whiteQueen;
+            this.blackQueen = blackQueen;
+        }
+    }
+
+    private boolean areQueensInSameRow(){
+        return whiteQueen.getRow() == blackQueen.getRow();
+    }
+
+    private boolean areQueensInSameColumn(){
+        return whiteQueen.getColumn() == blackQueen.getColumn();
+    }
+
+    private boolean areQueensInSameDiagonal(){
+        return (whiteQueen.getRow() - whiteQueen.getColumn()) == (blackQueen.getRow() - blackQueen.getColumn()) ||
+                (whiteQueen.getRow() + whiteQueen.getColumn()) == (blackQueen.getRow() + blackQueen.getColumn());
+    }
+
+    boolean canQueensAttackOneAnother() {
+        return areQueensInSameRow() || areQueensInSameColumn() || areQueensInSameDiagonal();
+    }
```

Comment:
Do you realize now after making a method for each check, you don't need your comments.
Here is a tip: If you are tempted to comment to explain sth, probably your code has a flaw in design. It is not always true. Sometimes, the problem is inherently complex. There you can comment to explain.
If the problem is not inherently complex but it is not clear for the reviewer, it means the coder introduced the complexity.
Try to make it simple to understand. Comment should explain why you are doing sth not what you are doing. To explain what you are doing use good namings like above ('areQueensInSameRow' etc).